src/app/services/data.service.ts

```diff
@@ -8,20 +11,117 @@ const STORAGE_KEY = 'TEST';
 })
 export class DataService {
 
-  items: any;
-
   constructor(@Inject(LOCAL_STORAGE) private storage: StorageService) {
-    this.getItems();
-    this.addItem('Christopher');
+    this.getStorage();
+  }
+
+  data: Storage;
+
+  static defaultStorage(): Storage {
+    return JSON.parse(JSON.stringify({
+      version: 0,
+      name: 'User',
+      setupCompleted: false,
+      co2: {
+        mode: 'average',
+        value: 148
+      },
+      vehicles: {
+        walking: {
+          id: 'walking',
+          name: 'Walking',
+          type: 'walking',
+          co2: 0,
+          active: true,
+          default: true,
+          travelmode: 'walking'
+        },
+        bicycling: {
+          id: 'bicycling',
+          name: 'Bicycle',
+          type: 'bicycle',
+          co2: 0,
+          active: true,
+          default: true,
+          travelmode: 'bicycling'
+        },
+        transit: {
+          id: 'transit',
+          name: 'Public Transportation',
+          type: 'train',
+          co2: 50,
+          active: true,
+          default: true,
+          travelmode: 'transit'
+        },
+        driving: {
+          id: 'driving',
+          name: 'Average Car',
+          type: 'car',
+          co2: 251,
+          active: true,
+          default: true,
+          travelmode: 'driving'
+        }
+      },
+      routes: {}
+    }));
+  }
+
+  static uuidv4(): string {
+    return (`${1e7}-${1e3}-${4e3}-${8e3}-${1e11}`).replace(/[018]/g, (c: any) =>
+      // tslint:disable-next-line:no-bitwise
+      (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
+    );
+  }
+
+  // Vehicles
+
+  public setVehicle(vehicle: Vehicle): void {
+    if (vehicle.id === 'new') {
+      vehicle.id = DataService.uuidv4();
+    }
+    this.data.vehicles[vehicle.id] = vehicle;
+    this.setStorage();
+  }
+
+  public getVehicle(vehicleId: string): Vehicle {
+    return this.data.vehicles[vehicleId];
+  }
+
+  public getVehicles(): Vehicle[] {
+    return Object.values(this.data.vehicles);
+  }
+
+  // CO2
+
+  public setCo2(co2: Co2): void {
+    this.data.co2 = co2;
+    this.setStorage();
+  }
+
+  public getCo2(): Co2 {
+    return this.data.co2;
+  }
+
+  // Storage
+
+  public setStorage(): void {
```

Comment:
Mh, its rather saveDataInStorage or something like that, isn't it? But yeah, I get why you choose that name^^

Reply:
would like to keep it that way :)